matrix.md

```diff
@@ -0,0 +1,136 @@
+| Primary use case | Untrusted library isolation / Secret/key protection / Fault isolation / Supply chain defense / Other: ___ | |
+| Subject selection | Code-centric / Data-centric / Hybrid | |
+| Finest isolation granularity | Function / Library / Thread / Process / VM / Other: ___ | |
+| **Isolation Approach** | | |
+| Hardware primitive(s) | MPK/PKU / MTE / CHERI / TEEs / Other: ___ / None | |
+| Software technique(s) | SFI / Boundary wrappers/marshalling / Memory-safe language / Language runtime / Other: ___ / None | |
+| Isolation abstraction | Process / Thread / Protection domain / VM / Container / Other: ___ | |
+| Requires runtime software support | ✅ / ❌ (if yes, describe: ___) | |
+| **Properties Enforced** | | |
+| Security properties provided | [Describe: e.g., "Memory integrity and confidentiality, syscall filtering"] | |
+| **Resilience** | | |
+| Compartment crashes isolated | ✅ / ❌ | |
+| **Interface Safety** | | |
+| Boundaries are checked/validated | ✅ / ❌ / Partial | |
+| **Trust & Threats** | | |
+| TCB includes | Compiler / OS / Firmware / CPU / Other: ___ (list applicable) | |
+| TCB approximate size | Small / Medium / Large / Very Large or [LOC/component count if known] | |
+| Side-channel resistance | None / Partial / Strong (if Partial/Strong, describe mitigations: ___) | |
+
+---
+
+## Runtime Efficiency
+
+| **Dimension** | **Metric** | **Value / Notes** |
+|----------------|------------|-------------------|
+| General runtime overhead | [%] or Low / Medium / High | |
+| Domain switch cost | [µs] or <1µs / 1-10µs / 10-100µs / >100µs | |
+| Domain creation cost | [ms] or <1ms / 1-10ms / >10ms | |
+| Inter-domain call latency | [µs] or Low / Medium / High | |
+| Inter-domain throughput | [MB/s] or Low / Medium / High | |
+| Memory overhead per domain | [KB/MB] or <10KB / 10-100KB / >100KB | |
+| Maximum domains (practical limit) | [number] or <10 / 10-100 / 100-1000 / >1000 | |
+| Performance scales with domain count | Well / Moderately / Poorly | |
+
+---
+
+## Usability / Adoptability
+
+| **Dimension** | **Metric** | **Value / Notes** |
+|----------------|------------|-------------------|
+| **Deployment Assumptions** | | |
+| What hardware does it need? | Commodity / Specialized (specify: ___) | |
+| What OS/kernel does it need? | Stock / Module / Modified / Custom | |
+| What privileges does it need? | User / Root / Kernel access | |
+| Other deployment requirements | [e.g., "Custom libc", "Modified toolchain"] | |
+| **Software Compatibility** | | |
+| What application changes are needed? | None / Annotations / API changes / Refactoring / Rewrite | |
+| Can it run existing binaries? | Yes / Recompile only / Source changes needed | |
+| What languages does it support? | C/C++ / Rust / Go / Managed / Any / Other: ___ | |
+| Other compatibility notes | [Describe any other constraints] | |
+| **Developer Effort** | | |
+| Porting effort for typical app | [person-days] or Low / Medium / High | |
```

Comment:
I am afraid that low/medium/high will not be super useful as we all have different references. Person-days measurements are a little better but still really hard to get precise... are we talking about an expert compartmentalizing, or someone who is compartmentalizing their first app? Also the effort will vary hugely depending on the structure of the app itself. It would be great to specify one particular app or software example to have a common reference, but there will be no such reference since compartmentalization frameworks may target different kinds of software.